pinot-core/src/main/java/org/apache/pinot/core/routing/RouteManager.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.routing;
+
+import java.util.Map;
+import org.apache.pinot.core.transport.ServerInstance;
+
+/**
+ * The {@code RouteManager} provides the routing information for a query that requests access to a Pinot table.
```

Comment:
We can't reuse / extend the existing RoutingManager / RoutingTable interfaces ?

Reply:
we can't do so because we will incur a reverted dependency from pinot-query-planner to pinot-broker (where RoutingManager/Table reside currently).